x-pack/platform/plugins/shared/onechat/public/application/components/conversations/conversation_rounds/chat_message_text.tsx

```diff
@@ -22,109 +19,48 @@ import {
   getDefaultEuiMarkdownParsingPlugins,
   getDefaultEuiMarkdownProcessingPlugins,
 } from '@elastic/eui';
+import { type PluggableList } from 'unified';
+import type { ConversationRoundStep } from '@kbn/onechat-common';
+import { useOnechatServices } from '../../../hooks/use_onechat_service';
+import {
+  Cursor,
+  esqlLanguagePlugin,
+  getVisualizationHandler,
+  loadingCursorPlugin,
+  visualizationPlugin,
+} from './markdown_plugins';
 
 interface Props {
   content: string;
+  stepsFromPrevRounds: ConversationRoundStep[];
+  stepsFromCurrentRound: ConversationRoundStep[];
 }
 
-const cursorCss = css`
-  @keyframes blink {
-    0% {
-      opacity: 0;
-    }
-    50% {
-      opacity: 1;
-    }
-    100% {
-      opacity: 0;
-    }
-  }
-
-  animation: blink 1s infinite;
-  width: 10px;
-  height: 16px;
-  vertical-align: middle;
-  display: inline-block;
-  background: rgba(0, 0, 0, 0.25);
-`;
-
-const Cursor = () => <span key="cursor" className={classNames(cursorCss, 'cursor')} />;
-
-const CURSOR = ` ᠎  `;
-
-const loadingCursorPlugin = () => {
-  const visitor = (node: Node, parent?: Parent) => {
-    if ('children' in node) {
-      const nodeAsParent = node as Parent;
-      nodeAsParent.children.forEach((child) => {
-        visitor(child, nodeAsParent);
-      });
-    }
-
-    if (node.type !== 'text' && node.type !== 'inlineCode' && node.type !== 'code') {
-      return;
-    }
-
-    const textNode = node as Text | InlineCode | Code;
-
-    const indexOfCursor = textNode.value.indexOf(CURSOR);
-    if (indexOfCursor === -1) {
-      return;
-    }
-
-    textNode.value = textNode.value.replace(CURSOR, '');
-
-    const indexOfNode = parent!.children.indexOf(textNode);
-    parent!.children.splice(indexOfNode + 1, 0, {
-      type: 'cursor' as Text['type'],
-      value: CURSOR,
-    });
-  };
-
-  return (tree: Node) => {
-    visitor(tree);
-  };
-};
-
-const esqlLanguagePlugin = () => {
-  const visitor = (node: Node, parent?: Parent) => {
-    if ('children' in node) {
-      const nodeAsParent = node as Parent;
-      nodeAsParent.children.forEach((child) => {
-        visitor(child, nodeAsParent);
-      });
-    }
-
-    if (node.type === 'code' && node.lang === 'esql') {
-      node.type = 'esql';
-    } else if (node.type === 'code') {
-      // switch to type that allows us to control rendering
-      node.type = 'codeBlock';
-    }
-  };
-
-  return (tree: Node) => {
-    visitor(tree);
-  };
-};
-
 /**
  * Component handling markdown support to the assistant's responses.
  * Also handles "loading" state by appending the blinking cursor.
  */
-export function ChatMessageText({ content }: Props) {
+export function ChatMessageText({ content, stepsFromPrevRounds, stepsFromCurrentRound }: Props) {
```

Comment:
NIT: That kind of dependencies on steps from all round feels wrong to me? I get why we technically need it to access the tool calls/results, but it feels super wrong.